Lib9c/Action/ClaimItems.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bencodex.Types;
+using Libplanet.Action;
+using Libplanet.Action.State;
+using Libplanet.Crypto;
+using Libplanet.Types.Assets;
+using Nekoyume.Extensions;
+using Nekoyume.Model.Item;
+using Nekoyume.Model.State;
+using Nekoyume.TableData;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Action
+{
+    public class ClaimItems : GameAction, IClaimItems
+    {
+        private const string ActionTypeText = "claim_item";
+
+        public Address AvatarAddress { get; private set; }
+        public IEnumerable<FungibleAssetValue> Amounts { get; private set; }
+
+        public ClaimItems() {}
+
+        public ClaimItems(Address avatarAddress, IEnumerable<FungibleAssetValue> amounts)
+        {
+            AvatarAddress = avatarAddress;
+            Amounts = amounts;
+        }
+
+        protected override IImmutableDictionary<string, IValue> PlainValueInternal =>
+            ImmutableDictionary<string, IValue>.Empty
+                .Add(AvatarAddressKey, AvatarAddress.Serialize())
+                .Add(AmountKey, new List(Amounts.Select(x => x.Serialize())));
+
+        protected override void LoadPlainValueInternal(
+            IImmutableDictionary<string, IValue> plainValue)
+        {
+            AvatarAddress = plainValue[AvatarAddressKey].ToAddress();
+            Amounts = ((List)plainValue[AmountKey]).Select(x => x.ToFungibleAssetValue());
+        }
+
+        public override IAccount Execute(IActionContext context)
+        {
+            context.UseGas(1);
+
+            var states = context.PreviousState;
+            var addressesHex = GetSignerAndOtherAddressesHex(context, AvatarAddress);
+
+            var avatarState = states.GetAvatarState(AvatarAddress);
+            if (avatarState is null)
+            {
+                throw new FailedLoadStateException(
+                    ActionTypeText,
+                    addressesHex,
+                    typeof(AvatarState),
+                    AvatarAddress);
+            }
```

Comment:
At first, I tried this way to get avatarState: `TryGetAvatarStateV2`. But we can't use signer address as agentAddress because the signer and avatarAddress can be different in this case. Is there some way to get agentAddress with avatarAddress?